test/gtest/unit/query_mem.cpp

```diff
@@ -0,0 +1,250 @@
+ */
+
+#include <gtest/gtest.h>
+#include <fstream>
+#include <filesystem>
+#include <optional>
+#include "file/file_utils.h"
+#include "nixl.h"
+#include "nixl_descriptors.h"
+
+class QueryMemTest : public ::testing::Test {
+protected:
+    void
+    SetUp() override {
+        // Create temporary test files
+        test_file1 = "/tmp/test_query_mem_1.txt";
+        test_file2 = "/tmp/test_query_mem_2.txt";
+        non_existent_file = "/tmp/non_existent_file.txt";
+
+        // Create test file 1
+        {
+            std::ofstream file(test_file1);
+            file << "Test content for file 1" << std::endl;
+            file.close();
+        }
+
+        // Create test file 2
+        {
+            std::ofstream file(test_file2);
+            file << "Test content for file 2" << std::endl;
+            file.close();
+        }
+    }
+
+    void
+    TearDown() override {
+        // Clean up test files
+        std::filesystem::remove(test_file1);
+        std::filesystem::remove(test_file2);
+    }
+
+    std::string test_file1;
+    std::string test_file2;
+    std::string non_existent_file;
+};
+
+TEST_F(QueryMemTest, QueryFileInfoListWithEmptyVector) {
+    std::vector<std::string> filenames;
+    std::vector<nixl_query_resp_t> resp;
+
+    nixl_status_t status = nixl::queryFileInfoList(filenames, resp);
+    EXPECT_EQ(status, NIXL_SUCCESS);
+    EXPECT_EQ(resp.size(), 0);
+}
+
+TEST_F(QueryMemTest, QueryFileInfoListWithEmptyFilenames) {
+    std::vector<std::string> filenames = {"", "", ""};
+    std::vector<nixl_query_resp_t> resp;
+
+    nixl_status_t status = nixl::queryFileInfoList(filenames, resp);
+    EXPECT_EQ(status, NIXL_SUCCESS);
+    EXPECT_EQ(resp.size(), 3);
+    EXPECT_FALSE(resp[0].has_value());
+    EXPECT_FALSE(resp[1].has_value());
+    EXPECT_FALSE(resp[2].has_value());
+}
+
+TEST_F(QueryMemTest, QueryFileInfoListWithMultipleExistingFiles) {
+    std::vector<std::string> filenames = {test_file1, test_file2};
+    std::vector<nixl_query_resp_t> resp;
+
+    nixl_status_t status = nixl::queryFileInfoList(filenames, resp);
+    EXPECT_EQ(status, NIXL_SUCCESS);
+    EXPECT_EQ(resp.size(), 2);
+    EXPECT_TRUE(resp[0].has_value());
+    EXPECT_TRUE(resp[1].has_value());
+    EXPECT_TRUE(resp[0].value().find("size") != resp[0].value().end());
+    EXPECT_TRUE(resp[1].value().find("size") != resp[1].value().end());
+}
+
+TEST_F(QueryMemTest, QueryFileInfoListWithMixedFiles) {
+    std::vector<std::string> filenames = {test_file1, non_existent_file, test_file2, ""};
+    std::vector<nixl_query_resp_t> resp;
+
+    nixl_status_t status = nixl::queryFileInfoList(filenames, resp);
+    EXPECT_EQ(status, NIXL_SUCCESS);
+    EXPECT_EQ(resp.size(), 4);
+    EXPECT_TRUE(resp[0].has_value()); // test_file1 exists
+    EXPECT_FALSE(resp[1].has_value()); // non_existent_file doesn't exist
+    EXPECT_TRUE(resp[2].has_value()); // test_file2 exists
+    EXPECT_FALSE(resp[3].has_value()); // empty filename doesn't exist
+}
+
```

Comment:
These are great! But all the following ones don't look like _unit_ tests to me since they use whole NIXL stack. Consider moving them from unit to functional tests in test/gtest.